core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -80,59 +113,103 @@ public int compareTo(TermSource o) {
       // sorted after they have been determined to be valid.
       return this.iter.getTopKey().compareColumnQualifier(o.iter.getTopKey().getColumnQualifier());
```

Comment:
You could add a comment to TermSource stating that the CQ is the docID for the term and is used for sorting.  Then note in the class comments that the inner class TermSource, is sortable by docID but this implementation is optimized and returns an unsorted list.